docs/project/contribution_tools.md

```diff
@@ -364,6 +365,22 @@ Our recommended way of installing is to use
 > still be used for other purposes, such as editing files, without interfering
 > with `bazel`.
 
+#### DevContainer
+
+To support developers join the project without deploying the build env, we
+provide VSCode `DevContainer`.
+
+-   Install VSCode and Docker;
```

Comment:
Just noting, I think we should work on the doc formatting (these should really be written more as sentences, 1. 2. 3., instead of a bullet sequence with semicolons). But I feel awkward having forgotten this PR for a week so I'm just going to approve, and worry about the docs later.